internal/client/policy/policy_resource.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package policyclient
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/client/transport"
+	policyclustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/policy/cluster"
+	policyclustergroupmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/policy/clustergroup"
+	policyorganizationmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/policy/organization"
+	policyworkspacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/policy/workspace"
+)
+
+// New creates a new policy resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for policy resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	ManageV1alpha1ClusterPolicyResourceServiceCreate(request *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyRequest) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyResponse, error)
+
+	ManageV1alpha1ClusterPolicyResourceServiceDelete(fn *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyFullName) error
+
+	ManageV1alpha1ClusterPolicyResourceServiceGet(fn *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyFullName) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyGetPolicyResponse, error)
+
+	ManageV1alpha1ClusterPolicyResourceServiceUpdate(request *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyRequest) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyResponse, error)
+
+	ManageV1alpha1ClustergroupPolicyResourceServiceCreate(request *policyclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupPolicyPolicyRequest) (*policyclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupPolicyPolicyResponse, error)
```

Comment:
nit: Clustergroup => ClusterGroup

Reply:
This is as per the generated code, should I still change it?

Reply:
Let it as per the generated client